### PR TITLE
Paytrace actions, refactored @isaiahtaylor's PR

### PR DESCRIPTION
**Pull request by feyzullah**
Refactored @isaiahtaylor's [PR](https://github.com/PipedreamHQ/pipedream/pull/2775)
To QA reviewer, I could not create a sandbox or formal account on Paytrace, just refactored the code already written, I hope it works well

<a href="https://gitpod.io/#https://github.com/PipedreamHQ/pipedream/pull/3431"></a>

### CODE REVIEW

**Comment by dylburger** [comment]
Thanks for submitting this PR! When we review PRs, we follow the [Pipedream component guidelines](https://pipedream.com/docs/components/guidelines/). If you're not familiar, here's a quick checklist:

- [ ] [Create components to address specific use cases whenever possible](https://pipedream.com/docs/components/guidelines/#component-scope)
- [ ] [Component `key`s should follow the format `app_name_slug-slugified-component-name`](https://pipedream.com/docs/components/guidelines/#component-key-pattern)
- [ ] Components should follow the [standard directory structure](https://pipedream.com/docs/components/guidelines/#folder-structure)
- [ ] [Prefer Node.js client libraries to REST APIs](https://pipedream.com/docs/components/guidelines/#using-apis-vs-client-libraries)
- [ ] When making API requests, [handle pagination](https://pipedream.com/docs/components/guidelines/#pagination) to ensure all data / events are processed
- [ ] [Use `secret` props](https://pipedream.com/docs/components/guidelines/#capturing-sensitive-data) to capture sensitive data
- [ ] Props and methods should be defined in [app files](https://pipedream.com/docs/components/guidelines/#app-files) whenever possible
- [ ] Document methods with [JS Docs](https://pipedream.com/docs/components/guidelines/#methods)
- [ ] [Use `optional` props](https://pipedream.com/docs/components/guidelines/#optional-vs-required-props) whenever possible, and [set a `default` value](https://pipedream.com/docs/components/guidelines/#default-values) where you can
- [ ] Use [async options](https://pipedream.com/docs/components/guidelines/#async-options) to accept user input wherever possible

File: components/paytrace/actions/batch-summary/batch-summary.mjs [hunk 1]
```diff
@@ -0,0 +1,29 @@
+import paytrace from "../../paytrace.app.mjs";
+
+export default {
+  name: "Batch Summary",
+  description: "This method can be used to export a summary of specific batch details or currently pending settlement details by card and transaction type.  If no optional parameter is provided, the latest batch details will be returned.",
```
Comment (lcaresia) [inline components/paytrace/actions/batch-summary/batch-summary.mjs:5]: Add docs link in the description.

You can do this for the other actions too.

```suggestion
  description: "This method can be used to export a summary of specific batch details or currently pending settlement details by card and transaction type.  If no optional parameter is provided, the latest batch details will be returned. [See docs here](<docs_link>)",
```

File: components/paytrace/actions/list-batch-transactions/list-batch-transactions.mjs [hunk 2]
```diff
@@ -0,0 +1,28 @@
+import paytrace from "../../paytrace.app.mjs";
+
+export default {
+  name: "List Batch Transactions",
+  description: "This method can be used to export settled transaction details within a specific batch.  This method will return one or more transaction records.",
```
Comment (lcaresia) [inline components/paytrace/actions/list-batch-transactions/list-batch-transactions.mjs:5]: Add docs link in the description.

File: components/paytrace/actions/list-batches/list-batches.mjs [hunk 3]
```diff
@@ -0,0 +1,35 @@
+import paytrace from "../../paytrace.app.mjs";
+
+export default {
+  name: "List Batches by Date Range",
+  description: "This method can be used to export a set of batch summary details with a provided date range.  This method will return one or more batch summary records.",
```
Comment (lcaresia) [inline components/paytrace/actions/list-batches/list-batches.mjs:5]: Add docs link in the description.

File: components/paytrace/actions/list-transactions/list-transactions.mjs [hunk 4]
```diff
@@ -0,0 +1,80 @@
+import paytrace from "../../paytrace.app.mjs";
+
+export default {
+  name: "List Transactions",
+  description: "This method can be used to export a set of credit card transaction details with a provided date range.  You can optimize your search by providing optional parameters.",
```
Comment (lcaresia) [inline components/paytrace/actions/list-transactions/list-transactions.mjs:5]: Add docs link in the description.

File: components/paytrace/actions/list-transactions/list-transactions.mjs [hunk 5]
```diff
@@ -0,0 +1,80 @@
...
+      label: "Transaction ID",
+      description: "A unique identifier for each transaction in the PayTrace system.",
+      optional: true,
+    },
+    transactionType: {
+      type: "string",
+      label: "Transaction Type",
+      description: "The transaction type to find transactions.",
+      optional: true,
+      options: [
```
Comment (lcaresia) [inline components/paytrace/actions/list-transactions/list-transactions.mjs:34]: Move this for a constants file.

E.g. `common/constants.mjs`

File: components/paytrace/paytrace.app.mjs [hunk 6]
```diff
@@ -1,11 +1,78 @@
...
   methods: {
-    // this.$auth contains connected account data
-    authKeys() {
-      console.log(Object.keys(this.$auth));
+    _getUrl(path) {
+      return `https://api.paytrace.com/v1${path}`;
+    },
+    _getHeaders(headers = {}) {
+      return {
+        "Authorization": `Bearer ${this.$auth.oauth_access_token}`,
```
Comment (lcaresia) [inline components/paytrace/paytrace.app.mjs:29]: Move this for a getter method.

```suggestion
        "Authorization": `Bearer ${this._accessToken()}`,
```

**Review by lcaresia — CHANGES_REQUESTED** [review]
Thanks for your contribution!

I requested just a little changes.

**Review by lcaresia — APPROVED** [review]
LGTM!

**Comment by vunguyenhung** [comment]
Hello @dannyroosevelt, I'm also can not create personal account on PayTrace. I think we can either get the account as Pipedream org, or merge this PR and monitor. What do you think?

**Comment by dannyroosevelt** [comment]
> Hello @dannyroosevelt, I'm also can not create personal account on PayTrace. I think we can either get the account as Pipedream org, or merge this PR and monitor. What do you think?

Go ahead and publish, and we'll see if we can get help QA'ing from the community. Thanks, Leo!

**Pull request merged**